compiler/rustc_codegen_ssa/src/debuginfo/type_names.rs

```diff
@@ -282,6 +282,17 @@ pub fn push_debuginfo_type_name<'tcx>(
             output.push_str("enum$<");
             push_item_name(tcx, def.did, true, output);
             push_type_params(tcx, substs, output, visited);
+
+            if let Variants::Single { index: variant_idx } = &layout.variants {
+                // Uninhabited enums can't be constructed and should never need to be visualized so
```

Comment:
What happens if these end up being behind a pointer? As in: `&() as as *const _ as *const Void`.

Reply:
Good question! In that case, we just don't have any information to display. 

![image](https://user-images.githubusercontent.com/831192/123660701-e399b380-d801-11eb-818c-dac86dbe6c85.png)